### PR TITLE
MenuItem detail Icon Fix

### PR DESCRIPTION
**Pull request by dbchristopher**
### :sparkles: Changes

- Deletes hard-coded MenuItem -> Icon color 

### :white_check_mark: Requirements

- [ ] Includes test coverage for all changes
- [x] Build and tests are passing
- [ ] Update documentation
- [x] Updated CHANGELOG
- [ ] Checked for i18n impacts
- [ ] Checked for a11y impacts
- [x] PR is ideally < ~400LOC

### :camera: Screenshots
Screen_Shot_2020-09-09_at_11_41_05_AM

### CODE REVIEW

File: packages/components/src/Menu/MenuItemLayout.tsx [hunk 1]
```diff
@@ -114,7 +114,6 @@ export const MenuItemLayout = styled(MenuItemWrapper)`
   `}
 
   ${Icon} {
-    color: ${({ theme: { colors } }) => colors.text1};
```
Comment (dbchristopher) [inline packages/components/src/Menu/MenuItemLayout.tsx:117]: @mdodgelooker it turns out the Icon default color was already `text1`. So the ultimate fix was just to delete this line 🥳

File: packages/components/src/Form/Inputs/Checkbox/__snapshots__/Checkbox.test.tsx.snap [hunk 2]
```diff
@@ -322,7 +322,7 @@ exports[`Checkbox should accept disabled & checked="mixed" 1`] = `
 >
   <input
     aria-checked="mixed"
-    checked={true}
+    checked={false}
```
Comment (mdodgelooker) [inline packages/components/src/Form/Inputs/Checkbox/__snapshots__/Checkbox.test.tsx.snap:325]: So odd... this snapshot passed for me, but then failed after I ran `yarn`.

Comment (dbchristopher) [reply packages/components/src/Form/Inputs/Checkbox/__snapshots__/Checkbox.test.tsx.snap:325]: oh dang! looks like I needed to run yarn too. I'll upload some new snapshots...

File: packages/components/src/Menu/MenuItemLayout.tsx [hunk 3]
```diff
@@ -124,10 +123,6 @@ export const MenuItemLayout = styled(MenuItemWrapper)`
   &[aria-current='true'] {
     background: ${({ theme: { colors } }) => colors.ui1};
     color: ${({ theme: { colors } }) => colors.text5};
-
-    ${Icon} {
-      color: ${({ theme: { colors } }) => colors.text1};
-    }
```
Comment (mdodgelooker) [inline packages/components/src/Menu/MenuItemLayout.tsx:130]: I think the one for `[disabled]` down on line 150 can also be removed.

**Review by mdodgelooker — CHANGES_REQUESTED** [review]
Looks good! Just one more `text1` to remove and then the example in playground should either move to `Menu.stories.tsx` or be deleted before merging.

File: packages/components/src/Menu/MenuItemLayout.tsx (hunk 3 above)
Comment (dbchristopher) [reply packages/components/src/Menu/MenuItemLayout.tsx:130]: oh good catch!

**Comment by dbchristopher** [comment]
Ok @mdodgelooker I updated the menu storybook and reverted the playground changes. this should be ready for final ✅

**Review by mdodgelooker — APPROVED** [review]
LGTM!

**Pull request merged**